File: careerforge/src/main/java/com/careerforge/blog/service/BlogService.java
package com.careerforge.blog.service;

import com.careerforge.blog.dto.BlogDto;
import com.careerforge.blog.dto.CreateBlogRequest;
import com.careerforge.blog.dto.RejectBlogRequest;
import com.careerforge.blog.dto.UpdateBlogRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface BlogService {
    BlogDto createStudentBlog(CreateBlogRequest request, UUID studentId);

    BlogDto createAdminBlog(CreateBlogRequest request, UUID adminId);

    BlogDto updateStudentBlog(UUID blogId, UpdateBlogRequest request, UUID studentId);

    Page<BlogDto> getPublishedBlogs(Pageable pageable);

    Page<BlogDto> getMyBlogs(UUID authorId, Pageable pageable);

    Page<BlogDto> getPendingBlogs(Pageable pageable);

    BlogDto approveBlog(UUID blogId, UUID adminId);

    BlogDto rejectBlog(UUID blogId, RejectBlogRequest request, UUID adminId);

    BlogDto getBlogById(UUID id);
}

File: careerforge-frontend/src/services/blogService.ts
import { apiClient } from '../config/api';
import { Blog, CreateBlogRequest, UpdateBlogRequest, RejectBlogRequest } from '../types/blog';

// We will return the data property directly from the backend's ResponseEntity wrapper
export const blogService = {
  // Student Endpoints
  async createStudentBlog(data: CreateBlogRequest): Promise<Blog> {
    const response = await apiClient.post('/api/student/blogs', data);
    return response.data.data;
  },

  async updateStudentBlog(id: string, data: UpdateBlogRequest): Promise<Blog> {
    const response = await apiClient.put(`/api/student/blogs/${id}`, data);
    return response.data.data;
  },

  async getMyBlogs(page: number, size: number): Promise<any> {
    const response = await apiClient.get('/api/student/blogs/my', {
      params: { page, size }
    });
    return response.data.data;
  },

  async getPublishedBlogs(page: number, size: number): Promise<any> {
    const response = await apiClient.get('/api/student/blogs', {
      params: { page, size }
    });
    return response.data.data;
  },

  async getBlogById(id: string): Promise<Blog> {
    const response = await apiClient.get(`/api/student/blogs/${id}`);
    return response.data.data;
  },

  // Admin Endpoints
  async createAdminBlog(data: CreateBlogRequest): Promise<Blog> {
    const response = await apiClient.post('/api/admin/blogs', data);
    return response.data.data;
  },

  async getPendingBlogs(page: number, size: number): Promise<any> {
    const response = await apiClient.get('/api/admin/blogs/pending', {
      params: { page, size }
    });
    return response.data.data;
  },

  async approveBlog(id: string): Promise<Blog> {
    const response = await apiClient.post(`/api/admin/blogs/${id}/approve`);
    return response.data.data;
  },

  async rejectBlog(id: string, data: RejectBlogRequest): Promise<Blog> {
    const response = await apiClient.post(`/api/admin/blogs/${id}/reject`, data);
    return response.data.data;
  }
};
